### DISCUSSION TITLE
Simulation control

### DISCUSSION THREAD

**Discussion by perceptualrobots** [cyberbotics/wrestling#55] [Q&A] [answered]
I would like to run the simulation many times, and looking to work out how to control the simulation, from my robot supervisor, so that it does the following:

- starts the simulation
- starts the robots
- resets the simulation
- resets the robots
- run without rendering
- run in computer time (not real time) without loop delays 

In https://cyberbotics.com/doc/reference/supervisor I see WB_SUPERVISOR_SIMULATION_MODE_FAST says "The simulation is running as fast as possible without the graphical renderings."

I did try this, however, I get this error, 

`'WrestlerSupervisor' object has no attribute 'WB_SUPERVISOR_SIMULATION_MODE_FAST'
`

Which would be the best commands to achieve the above functionality?

**Comment by JustLuckyRobotics** [comment]
id recommend getting affiliated with the files located somewhere similar to C:\Program Files\Webots\lib\controller\python\controller

**Comment by Jean-Eudes-le-retour** [comment, marked as answer]
The text descriptions in the documentation are mainly for the C programming language, but there are sections where you can select a programming language to see how the function is written in that language, see https://cyberbotics.com/doc/reference/supervisor?tab-language=python#wb_supervisor_simulation_set_mode

**Reply by perceptualrobots** [reply]
> The text descriptions in the documentation are mainly for the C programming language, but there are sections where you can select a programming language to see how the function is written in that language, see https://cyberbotics.com/doc/reference/supervisor?tab-language=python#wb_supervisor_simulation_set_mode

Thanks for your response, though this only addresses a small part of my query (so not sure why it has been marked as an answer). I did try SIMULATION_MODE_FAST though the graphical rendering was still happening, which is not what the documentation suggests, 

`"without the graphical renderings."` 

On one of my other points, currently I am having to manually click the play button in the GUI to kick-off each run. How can this be done programatically, I don't see any "Start" methods in the Supervisor documentation  [https://cyberbotics.com/doc/reference/supervisor](https://cyberbotics.com/doc/reference/supervisor)?

**Reply by Jean-Eudes-le-retour** [reply]
Oh I see, you can have more control on the Webots instance by using [the command-line argument](https://cyberbotics.com/doc/guide/starting-webots#command-line-arguments) when you launch the program.

In your case you should try `--no-rendering --batch --mode=fast` to directly launch a simulation in fast mode with no rendering. The `--batch` arguments prevents pop-ups from preventing a simulation to run.
